src/shadowbox/integration_test/test.sh

```diff
@@ -88,12 +88,13 @@ function test_with_retries() {
 
   for ((i=1; i<=${max_attempts}; i++))
   do
-    output=$(${test_function})
-    result_code=$?
-    if [[ $result_code -eq 0 ]]; then
+    # The script will exit if the below test returns with a non-zero status code
+    # (due to set -e), so we capture the result code separately.
+    $(${test_function}) && result_code=$? || result_code=$?
```

Comment:
Remove the `$()`. No need for that, the test_function call directly should work.

Reply:
I think you can also do this, which is clearer:
```bash
if ${test_function}; then
  return
else 
  result_code=$?
fi
```